blkid/internal/probe/probe.go

```diff
@@ -28,7 +28,13 @@ type Prober interface {
 	// Magic returns the magic value for the filesystem or volume manager.
 	Magic() []*magic.Magic
 	// Probe runs the further inspection and returns the result if successful.
-	Probe(Reader) (*Result, error)
+	Probe(Reader, magic.Magic) (*Result, error)
+}
+
+// Probe is a magic detection result.
+type Probe struct {
```

Comment:
just `Result` maybe, as it will be `probe.Result`

Reply:
We have result there below

Reply:
then we need a different name... certainly `Probe` is not a good name

Reply:
even `ProberWithMagic`, or `MagicMatch` is a better name